src/gui/guirequests.cpp

```diff
@@ -836,18 +837,28 @@ ExitCode GuiRequests::getAppState(AppStateKey key, QString &value) {
     ExitCode exitCode = ExitCodeUnknown;
     QDataStream resultStream(&results, QIODevice::ReadOnly);
     resultStream >> exitCode;
-    if (exitCode == ExitCodeOk) {
-        resultStream >> value;
+    if (exitCode != ExitCodeOk) {
+        return exitCode;
     }
 
+    QString valueQStr;
+    resultStream >> valueQStr;
+
+    if (std::string valueStr = valueQStr.toStdString(); !CommonUtility::stringToAppStateValue(valueStr, value)) {
+        return ExitCodeSystemError;
+    }
     return exitCode;
 }
 
-ExitCode GuiRequests::updateAppState(AppStateKey key, const QString &value) {
+ExitCode GuiRequests::updateAppState(AppStateKey key, const AppStateValue &value) {
     QByteArray params;
     QDataStream paramsStream(&params, QIODevice::WriteOnly);
     paramsStream << key;
-    paramsStream << value;
+    std::string valueStr = "";
```

Comment:
Not important but no need to initialize a std::string (its a class with a default constructor) with an empty string. You'll lose some time creating the right value then copying it to 'valueStr'.